ts/features/wallet/onboarding/bancomat/screens/SearchBankScreen.tsx

```diff
@@ -1,18 +1,137 @@
+import BaseScreenComponent from "../../../../../components/screens/BaseScreenComponent";
 import { GlobalState } from "../../../../../store/reducers/types";
+import IconFont from "../../../../../components/ui/IconFont";
+import { BankPreviewItem } from "../components/BankPreviewItems";
+import { abis } from "../mock/mockData";
+import {
+  cancelButtonProps,
+  confirmButtonProps
+} from "../../../../bonus/bonusVacanze/components/buttons/ButtonConfigurations";
+import I18n from "../../../../../i18n";
+import { navigateBack } from "../../../../../store/actions/navigation";
+import FooterWithButtons from "../../../../../components/ui/FooterWithButtons";
+import { withLightModalContext } from "../../../../../components/helpers/withLightModalContext";
+import { LightModalContextInterface } from "../../../../../components/ui/LightModal";
+import TosBonusComponent from "../../../../bonus/bonusVacanze/components/TosBonusComponent";
 
+type Props = LightModalContextInterface &
+  ReturnType<typeof mapStateToProps> &
+  ReturnType<typeof mapDispatchToProps>;
 /**
  * This screen allows the user to choose a specific bank to search for their Bancomat.
  * the user can also choose not to specify any bank and search for all Bancomat in his name
  * @constructor
  */
-export const SearchBankScreen: React.FunctionComponent = () => <View />;
+const SearchBankScreen: React.FunctionComponent<Props> = (props: Props) => {
+  const [searchText, setSearchText] = React.useState("");
+  const [filteredList, setFilteredList] = React.useState(
+    [] as typeof props.bankList
+  );
 
-const mapDispatchToProps = (_: Dispatch) => ({});
+  const [isSearchStarted, setIsSearchStarted] = React.useState(false);
 
-const mapStateToProps = (_: GlobalState) => ({});
+  const renderListItem = (isList: boolean) => (
+    info: ListRenderItemInfo<any>
+  ) => (
+    <BankPreviewItem
+      bank={info.item}
+      inList={isList}
+      onPress={props.onItemPress}
+    />
+  );
 
-export default connect(mapStateToProps, mapDispatchToProps)(SearchBankScreen);
+  const handleFilter = (text: string) => {
+    setIsSearchStarted(true);
+    const resultList = props.bankList.filter(
+      bank =>
+        bank.abi.toLowerCase().indexOf(text.toLowerCase()) > -1 ||
+        bank.name.toLowerCase().indexOf(text.toLowerCase()) > -1
+    );
+    setFilteredList(resultList);
+    setSearchText(text);
+  };
+
+  const keyExtractor = (bank: any): string => bank.abi;
+
+  const openTosModal = () => {
+    props.showModal(
+      <TosBonusComponent
+        tos_url={"https://google.com"}
+        onClose={props.hideModal}
+      />
+    );
+  };
+
+  return (
+    <BaseScreenComponent
+      goBack={true}
+      headerTitle={I18n.t("bonus.bpd.searchAbi.title")}
+    >
+      <SafeAreaView style={{ flex: 1 }}>
+        <Content style={{ flex: 1 }}>
+          {!isSearchStarted && (
+            <>
+              <View spacer={true} large={true} />
+              <Text>
```

Comment:
I guess we should use Body

```typescript
<Body>
	<Body weight={"Bold"}>
	  {I18n.t("bonus.bpd.searchAbi.description.text1")}
	</Body>
	{I18n.t("bonus.bpd.searchAbi.description.text2")}
	<Link onPress={openTosModal}>
	  {I18n.t("bonus.bpd.searchAbi.description.text3")}
	</Link>
</Body>
```

@fabriziofff if yes, Body doesn't allow "Bold" weight. Should it do it?

Reply:
@Undermaken At the moment `<Body>` doesn't allow the "Bold".
![Schermata 2020-10-02 alle 14 01 26](https://user-images.githubusercontent.com/26501317/94921048-c5439900-04b7-11eb-9a97-3c09ae4b3f5d.png)
I've used `<Label>` but we can discuss with the design if is intended or we should add it.

Reply:
I think in this case we can use <Body> for the normal text and <Label> for the bold. In revision phase we can ask if we should add the "Bold" to the <Body> and ask to update the invision library too. I'd like to have consistency between the invision library and our core components.